org.eclipse.jdt.debug.tests/tests/org/eclipse/jdt/debug/tests/eval/Java8Tests.java

```diff
@@ -241,4 +241,38 @@ public void testContextEvaluations() throws Exception {
 		}
 	}
 
+	public void testEvaluate_GH275_StaticBinaryMethod_EvaluateSnippetWithImportedTypes() throws Exception {
+		IJavaThread javaThread = null;
+		try {
+			IJavaLineBreakpoint bp = createLineBreakpoint(getType("com.debug.test.Observation"), 25);
+			javaThread = launchToLineBreakpoint("GH275", bp);
+			assertNotNull("The program did not suspend", javaThread);
+
+			String snippet = "new Observation((Object) subject, (Consumer) action)";
```

Comment:
@testforstephen isn't this the same snippet you are trying in Eclipse UI which fails for you ?

Reply:
The snippet is same, but the evaluation context is not same. In this test case, it's stopped at Line 25. The failure case I mentioned is stopped at Line 19. It's supposed to pass as well.